reporter/otlp_reporter.go

```diff
@@ -738,19 +743,26 @@ func createFunctionEntry(funcMap map[funcInfo]uint64,
 }
 
 // getSampleAttributes builds a sample-specific list of attributes.
-func getSampleAttributes(profile *profiles.Profile, k traceAndMetaKey) []uint64 {
+func getSampleAttributes(profile *profiles.Profile,
+	k traceAndMetaKey, attributeMap map[string]uint64) []uint64 {
 	indices := make([]uint64, 0, 4)
 
 	addAttr := func(k attribute.Key, v string) {
 		if v == "" {
 			return
 		}
-
-		indices = append(indices, uint64(len(profile.AttributeTable)))
+		attributeCompositeKey := string(k) + "_" + v
```

Comment:
Should we lift this inner function into a standalone function that can also be called from [getResource](https://github.com/open-telemetry/opentelemetry-ebpf-profiler/blob/a0b8ba402a37161732f3e63334699aeaa21f96d8/reporter/otlp_reporter.go#L457-L480) since attributes are also added there (in a separate attribute table)?